### ISSUE TITLE
workflow-controller panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by ThreadDao** [argoproj/argo-workflows#9062]
## Checklist

 

* [ ] Double-checked my configuration.
* [ ] Tested using the latest version.
* [ ] Used the Emissary executor.

## Summary
We faced the argo workflow controller panic problem:
```
time="2022-06-27T02:11:05.939Z" level=error msg="Mark error node" error="workflowtemplates.argoproj.io \"milvus-compaction-test-1\" not found" namespace=argo nodeName="milvus-compaction-cron-1-1656261000[0].compaction-test-cron.standalone-compaction-test[1].wait-healthy" workflow=milvus-compaction-cron-1-1656261000
time="2022-06-27T02:11:05.939Z" level=info msg="node milvus-compaction-cron-1-1656261000-1610435539 phase Running -> Error" namespace=argo workflow=milvus-compaction-cron-1-1656261000
time="2022-06-27T02:11:05.939Z" level=info msg="node milvus-compaction-cron-1-1656261000-1610435539 message: workflowtemplates.argoproj.io \"milvus-compaction-test-1\" not found" namespace=argo workflow=milvus-compaction-cron-1-1656261000
time="2022-06-27T02:11:05.939Z" level=info msg="node milvus-compaction-cron-1-1656261000-1610435539 finished: 2022-06-27 02:11:05.939283043 +0000 UTC" namespace=argo workflow=milvus-compaction-cron-1-1656261000
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1992bf6]

goroutine 351 [running]:
github.com/argoproj/argo-workflows/v3/workflow/templateresolution.(*Context).GetTemplateFromRef(0x0, 0xc0009ab350)
	/go/src/github.com/argoproj/argo-workflows/workflow/templateresolution/context.go:125 +0x36
github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).GetNodeTemplate(0xc0016606c0, 0xc000693b80)
	/go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:2093 +0x105
github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).assessNodeStatus(0xc0016606c0, 0xc00088a7e0, 0xc000693b80)
	/go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:1162 +0x9e
github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).podReconciliation.func1(0xc00088a7e0)
	/go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:997 +0x345
github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).podReconciliation.func2(0x23086b0)
	/go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:1037 +0x92
created by github.com/argoproj/argo-workflows/v3/workflow/controller.(*wfOperationCtx).podReconciliation
	/go/src/github.com/argoproj/argo-workflows/workflow/controller/operator.go:1035 +0x21a
```
What happened/what you expected to happen?

What version are you running?
Image: `quay.io/argoproj/workflow-controller:v3.3.6`

K8s version
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.4", GitCommit:"e6c093d87ea4cbb530a7b2ae91e54c0842d8308a", GitTreeState:"clean", BuildDate:"2022-02-16T12:32:02Z", GoVersion:"go1.17.7", Compiler:"gc", Platform:"linux/amd64"}
```

## Diagnostics

Paste the smallest workflow that reproduces the bug. We must be able to run the workflow.

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: workflow-controller-configmap
data:
  instanceID: my-ci-controller
  artifactRepository: |    # However, all nested maps must be strings
    archiveLogs: true
    s3:
      endpoint: xxx:xxx
      bucket: my-bucket
      insecure: true
      accessKeySecret:
        name: argo-artifacts
        key: accesskey
      secretKeySecret:
        name: argo-artifacts
        key: secretkey
  links: |
    - name: Pod Link
      scope: pod
      url: http://logging-facility?namespace=${metadata.namespace}&podName=${metadata.name}&startedAt=${status.startedAt}&finishedAt=${status.finishedAt}
    - name: Pod Logs Link
      scope: pod-logs
      url: http://logging-facility?namespace=${metadata.namespace}&podName=${metadata.name}&startedAt=${status.startedAt}&finishedAt=${status.finishedAt}
    - name: Event Source Logs Link
      scope: event-source-logs
      url: http://logging-facility?namespace=${metadata.namespace}&podName=${metadata.name}&startedAt=${status.startedAt}&finishedAt=${status.finishedAt}
    - name: Sensor Logs Link
      scope: sensor-logs
      url: http://logging-facility?namespace=${metadata.namespace}&podName=${metadata.name}&startedAt=${status.startedAt}&finishedAt=${status.finishedAt}
  persistence: |
    connectionPool:
      maxIdleConns: 100
      maxOpenConns: 0
      connMaxLifetime: 0s
    nodeStatusOffLoad: true
    archive: true
    archiveTTL: 180d
    mysql:
          host: mysql.argo.svc.cluster.local
          port: 3306
          database: argo
          tableName: argo_workflows
          userNameSecret:
            name: argo-mysql-config
            key: username
          passwordSecret:
            name: argo-mysql-config
            key: password
```
**workflow-controller log:**  [workflow-controller.log](https://github.com/argoproj/argo-workflows/files/8988248/workflow-controller.log)

```bash
# Logs from the workflow controller:
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}

# If the workflow's pods have not been created, you can skip the rest of the diagnostics.

# The workflow's pods that are problematic:
kubectl get pod -o yaml -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded

# Logs from in your workflow's wait container, something like:
kubectl logs -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

---

**Message from the maintainers**:

Impacted by this regression? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by sarabala1979** [comment]
@ThreadDao  fixed in https://github.com/argoproj/argo-workflows/pull/8961 
try with latest tag

**Comment by ThreadDao** [comment]
> @ThreadDao fixed in #8961 try with latest tag

Thanks a lot. the workflow controller pod restarted after deleting the running workflow. Next time I encounter this problem I will try the new version
